### PR TITLE
Implement Mail Send

### PR DESCRIPTION
**Pull request by andela-echigbo**
Why?
When a traveler completes the booking process, he should be notified via
of the status of the process.

How?
generate action mailer object,
write the controller and the views for the
write the mail views
add letter_opener to the gemfile to enable opening mail with it
bundle

Completing "Implement Mail Send"
https://www.pivotaltracker.com/story/show/110016366
[finished #110016366]

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,16 @@ class ApplicationController < ActionController::Base
...
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def authorize
+    redirect_to login_url, "Login to perform the action"  if current_user.nil?
+  end
+
+  helper_method :current_user
```
Comment (andela-iarisah) [inline app/controllers/application_controller.rb]: nice....it is however more readable to move `line 16` to `line 11`

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -6,8 +6,10 @@ def book
   end
 
   def create
-    book = Booking.new(booking_params)
+    book = Booking.make_new(booking_params)
```
Comment (andela-iarisah) [inline app/controllers/bookings_controller.rb]: I would like to know the difference between `new` and `make_new`

Comment (andela-iarisah) [inline app/controllers/bookings_controller.rb]: scratch that. I see a class method

File: app/views/application/_header.html.erb [hunk 3]
```diff
@@ -6,14 +6,14 @@
...
+          <li><a href="<%=  %>" class="btn auth-btn">signin</a></li>
+          <li><a href="<%=  %>" class="btn auth-btn">signup</a></li>
           <!-- <li><a href="mobile.html">Mobile</a></li> -->
         </ul>
         <ul class="side-nav" id="mobile-demo">
           <li><a href="<%= flights_path %>">All flight</a></li>
-          <li><a href="#!">signin</a></li>
-          <li><a href="#!">sginup</a></li>
+          <li><a href="<%=  %>">signin</a></li>
+          <li><a href="<%=  %>">signup</a></li>
```
Comment (andela-iarisah) [inline app/views/application/_header.html.erb]: empty interpolations? :-)

File: config/routes.rb [hunk 4]
```diff
@@ -1,8 +1,4 @@
 Rails.application.routes.draw do
-  # get "flights/index"
-
-  # get "flights/search"
-
```
Comment (andela-iarisah) [inline config/routes.rb]: :+1:

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (andela-echigbo) [inline app/controllers/application_controller.rb]: Well said. I have done that

File: app/controllers/bookings_controller.rb (hunk 2 above)
Comment (andela-echigbo) [inline app/controllers/bookings_controller.rb]: Yeah, I didn't know of `before_create` at the time of writing. I will change that tho. Thank you

File: app/views/application/_header.html.erb (hunk 3 above)
Comment (andela-echigbo) [inline app/views/application/_header.html.erb]: This was updated in a later commit. Thank you

**Pull request merged**